ignite/engine/events.py

```diff
@@ -437,7 +447,9 @@ def print_epoch(engine):
         # print_epoch handler is now unregistered
     """
 
-    def __init__(self, event_name: Union[CallableEventWithFilter, Enum, EventsList], handler: Callable, engine):
+    def __init__(
+        self, event_name: Union[CallableEventWithFilter, Enum, EventsList], handler: Callable, engine: Engine
```

Comment:
Do you think it would be better to add `EventType` as `Union[CallableEventWithFilter, Enum, EventsList, Events]` ?

Reply:
Do you mean to define a type alias?
```
EventType = TypeVar("EventType", CallableEventWithFilter, Enum, EventsList, Events)
```
or just add `Events` inside the Union block?

Yeah, it makes sense to add it regarding the purpose of the class. For `mypy` it doesn't matter with or without it 😄

Reply:
just a side note, it is definitely a bad idea to use a `TypeVar` because `mypy` doesn't understand the usage. directly complaining about it. There is also an open issue in the `mypy` project and it seems like, it works as designed, not a bug, it is a feature...

Reply:
Yes, the idea was to create an alias instead of passing all the times Union of bunch of classes.
Well, if it is not possible, OK, but sad